lib/praxis/response_definition.rb

```diff
@@ -238,17 +238,36 @@ def validate_headers!(response)
     def validate_content_type!(response)
       return unless media_type
 
-      extracted_identifier = nil
+      response_content_type = {}
       if response.headers['Content-Type']
-        extracted_identifier = Praxis::ContentTypeParser.parse(response.headers['Content-Type'])[:type]
+        response_content_type = Praxis::ContentTypeParser.parse(response.headers['Content-Type'])
       end
 
-      if media_type.identifier != extracted_identifier
```

Comment:
It occurs to me, that an easy way to get efficiency in the parsing, could be to add an instance method to both the `Response` and the `MediaType` classes that can get the pre-parsed pieces of the content-type/media_type.
These methods could cache the resulting hash (i.e. the typical @res ||= ... ) to avoid double parsing?
Maybe we could have:
```Response.parsed_content_type``` => hash
and ```MediaType```.parsed_identifier => hash
if we are cute, we should make sure every time we set the content-type header in a response or the identifier in a media-type, that we clear the cached parsed version.